### PR TITLE
Multiple emote suffix chain support

### PR DESCRIPTION
**Pull request by blushies**
also added `:flip`

### CODE REVIEW

File: assets/chat/js/formatters.js [hunk 1]
```diff
@@ -20,21 +20,25 @@ class EmoteFormatter {
...
-            this.regex = new RegExp(`(^|\\s)(${emoticons})(:(${suffixes}))?(?=$|\\s)`, 'gm');
+            this.regex = new RegExp(`(?:^|\\s)(${emoticons})((?::(?:${suffixes}))*)(?=$|\\s)`, 'gm');
         }
-        
-        return str.replace(this.regex, function(m) {
-            if (!m.includes(':')) {
-                const emote = m.replace(/\s/g, '');
+
+        return str.replace(this.regex, match => {
+            const index_of_suffix = match.indexOf(':');
```
Comment (KenanY) [inline assets/chat/js/formatters.js]: camelCase like other variables?

File: assets/chat/js/formatters.js [hunk 2]
```diff
@@ -20,21 +20,25 @@ class EmoteFormatter {
...
+                const emote = match.replace(/\s/g, '');
                 return ' <span title=' + emote + ' class="chat-emote chat-emote-' + emote + '">' + emote + '</span>';
             } else {
-                const emote = m.split(':')[0].replace(/\s/g, '');
-                const suffix = m.split(':')[1].replace(/\s/g, '');
-                return ' <span class="generify-container ' + GENERIFY_OPTIONS[suffix] + '"><span title="'+ m +'" class="chat-emote chat-emote-'+ emote +'">'+ m +' </span></span>';
+                const emote = match.slice(0, index_of_suffix).replace(/\s/g, '');
+                let suffixes = match.slice(index_of_suffix + 1).split(':').map(suffix => suffix.replace(/\s/g, ''));
+                // dedupe
+                suffixes = Object.keys(suffixes.reduce((acc, curr) => {
```
Comment (KenanY) [inline assets/chat/js/formatters.js]: should we just make `suffixes` a `Set` from the beginning?

File: assets/emotes.json [hunk 3]
```diff
@@ -1,138 +1,130 @@
-{
-    "destiny": [
```
Comment (KenanY) [inline assets/emotes.json]: Just making sure, is this restructuring necessary and/or okay?

File: assets/chat/js/chat.js [hunk 4]
```diff
@@ -425,15 +424,18 @@ class Chat {
...
+        this.emoticons = new Set(emotes);
+        this.isEmoteRegex = new RegExp(`^(${emotes.join('|')})((?::(?:${Object.keys(GENERIFY_OPTIONS).join('|')}))*)$`);
         return this;
     }
 
+    isEmote(text) {
+        if (!this.isEmoteRegex) {
+            return false;
+        }
+        return this.isEmoteRegex.test(text);
```
[the remark this answers is not in the document]
Comment (blushies) [reply assets/chat/js/chat.js]: I like my check more because its more readable

File: assets/emotes.json (hunk 3 above)
Comment (blushies) [reply assets/emotes.json]: yes, the app only uses the `destiny` property

File: assets/chat/js/formatters.js (hunk 1 above)
Comment (blushies) [reply assets/chat/js/formatters.js]: I don't think camel case is the standard here.

File: assets/chat/js/formatters.js (hunk 2 above)
Comment (blushies) [reply assets/chat/js/formatters.js]: Good idea.

**Pull request closed without merging**